src/util/asset_analysis.py

```diff
@@ -0,0 +1,63 @@
+"""Helper functions used to analyze data on assets."""
+
+from packaging.version import InvalidVersion, Version
+
+
+def is_older_than_version(
+        supported: list[str],
+        target_version: Version) -> bool:
+    """Determines if an asset that supports a list of versions is older than a
+    given target version.
+
+    Args:
+        supported: The list of versions an asset supports
+        target_version: The version to compare the supported versions to
+
+    Returns:
+        True if all supported versions are older than the target version, False
+        otherwise.
+    """
+    for v in supported:
+        try:
+            version = Version(v)
+            if version >= target_version:
+                return False
+        except InvalidVersion:
+            continue
+    return True
+
+
+def keyword_in_category_url(categories: list[dict], keyword: str) -> bool:
+    """Determines if a keyword exists in the path of at least one of the
+    category URLs for the given list of categories.
+
+    Args:
+        categories: The list of categories to parse the URLs of
+        keyword: The keyword to look for
+
+    Returns:
+        True if the keyword is found in at least one of the category URLs, False
+        otherwise.
+    """
+    for category in categories:
```

Comment:
`return any(keyword in category['url'] for category in categories)`

Here and in other functions, use a generator expression and any()/all()